raft/raft.go

```diff
@@ -726,12 +727,19 @@ func (r *raft) Step(m pb.Message) error {
 			// but it will not receive MsgApp or MsgHeartbeat, so it will not create
 			// disruptive term increases
 			r.send(pb.Message{To: m.From, Type: pb.MsgAppResp})
-		} else {
+			return nil
+		case m.Type == pb.MsgPreVote || m.Type == pb.MsgPreVoteResp:
+			// If we receive a MsgPreVote from a node with a lower term number,
+			// we reject it. For a MsgPreVoteResp we simply pass it to our
+			// stepFunc. For a stale MsgPreVoteResp (think of late responses,
+			// we've already transitioned to a leader or follower state) this
```

Comment:
But if we've transitioned to follower and then *back* to pre-candidate, there's no other Term check in StepCandidate. We could accept a MsgPreVoteResp for term T as a valid pre-vote in term T+1.